src/mylinef.cpp

```diff
@@ -290,19 +292,22 @@ MyLineF::intersects_crossHypot(const QLineF& l, QPointF* intersectionPoint) cons
 QLineF::IntersectionType
 MyLineF::intersects_flsiOrigX(const QLineF &l, QPointF *intersectionPoint) const
 {
+	constexpr qreal tolerance = 1000;
+
 	// ipmlementation is based on Graphics Gems III's "Faster Line Segment Intersection"
 	const QPointF a = p2() - p1();
 	const QPointF b = l.p1() - l.p2();
 	const QPointF c = p1() - l.p1();
 
-	const qreal denominator = std::fma(a.y(), b.x(), -a.x() * b.y());
+	const qreal axby = a.x() * b.y();
+	const qreal denominator = std::fma(a.y(), b.x(), -axby);
 	if (!std::isfinite(denominator))
 		return NoIntersection;
 
 	const qreal length = std::fma(a.x(), a.x(), a.y() * a.y());
-	const qreal epsilon = length * std::numeric_limits<qreal>::epsilon();
+	const qreal epsilon = tolerance * length * std::numeric_limits<qreal>::epsilon();
 
-	if (std::abs(denominator) < epsilon)  {
+	if (std::abs(denominator) < epsilon) [[unlikely]] {
```

Comment:
Are we okay with severely penalizing the parallel/colinear case with the `[[unlikely]]`?